app/controllers/users/news_controller.rb

```diff
@@ -6,6 +6,12 @@ def index
 
     def show
       @news = News.find(params[:id])
+      # news_userがあれば既読と判定する
+      unless current_user.news_users.present?
```

Comment:
unless文はパッとみてわかりにくいのでできる限りif文がいいですね。以下のように書くと素直に読みやすくなりそうです。
```suggestion
     if current_user.news_users.blank?
```

Reply:
そして以下のようにするといいかもですね。ぱっと見何をしてるのかわかりにくいので、privateメソッドに分割してそれをアクションないで呼び出すとかもありです。

```suggestion

def show
  @news = News.find(params[:id])
  update_read_status!
end

private
def update_read_status!
  if current_user.news_users.blank?
     news_user = current_user.news_users.build
     news_user.news = @news
     news_user.save!
  end
end
```

Reply:
showアクションは詳細ページを見せることが責務なので、それ以外の責務（今回で言うと既読管理）をshowアクション内に記述したくないと言う気持ちがあります。なのでこんな感じで記載したりします。